vortex-array/src/array/chunked/canonical.rs

```diff
@@ -210,19 +210,37 @@ fn pack_varbin(
     dtype: &DType,
     _nullability: Nullability,
 ) -> VortexResult<VarBinArray> {
-    let len = chunks.iter().map(|chunk| chunk.len()).sum();
-    let mut builder = VarBinBuilder::<i32>::with_capacity(len);
+    let len: usize = chunks.iter().map(|c| c.len()).sum();
+    let mut offsets = Vec::with_capacity(len + 1);
+    offsets.push(0);
+    let mut buffer = Vec::new();
 
     for chunk in chunks {
         let chunk = chunk.clone().into_varbin()?;
-        chunk.with_iterator(|iter| {
```

Comment:
While this method is convenient it's not very fast if you know you just want to copy the values